### PR TITLE
Session, cookies and security

### PR DESCRIPTION
**Pull request by enricostano**
### What is this?
We found out that people tend to leave their TO account logged in while using public computers. This is a security concern, of course.

We use cookies to store sessions, this is not the best solution since cookies can be stolen and we should switch to some database store for sessions (Memcache, Redis, etc). But that's out of the scope of this PR.

With this PR we're addressing:

1. Update Devise gem to get some important security update
2. Adopt the new Rails `secrets.yml` and remove a deprecation warning, more info [here](http://guides.rubyonrails.org/upgrading_ruby_on_rails.html#config-secrets-yml)
3. Add an expiration to the session cookie

### How to test it
1. Test that you must log in again after 1 hour you stopped using TO
2. Test that if you check the option `Remember me` during login it shouldn't ask to log in again during 2 weeks

### CODE REVIEW

**Comment by rewritten** [comment]
Maybe it can be tested using Time.travel_to (for the session duration) and abusing the test client so non-persistent cookies are deleted (to check for "remember me" tokens)

File: Gemfile.lock [hunk 1]
```diff
@@ -394,7 +394,7 @@ DEPENDENCIES
   coffee-rails
   dalli
   database_cleaner
-  devise
+  devise (= 3.5.6)
```
Comment (rewritten) [inline Gemfile.lock]: I prefer soft dependencies (`~> 3.5.6`) if possible.

**Comment by rewritten** [comment]
Are the three changes really related? Devise update, secret token configuration and session expiry? I don't see it.

**Comment by rewritten** [comment]
Final question: given that the session expiration is counted from the last request, isn't 1 hour a little too long? That is 1 hour of leaving the computer unattended.

File: Gemfile.lock (hunk 1 above)
Comment (enricostano) [inline Gemfile.lock]: Well, we'll need another consensus then. I've been bitten by soft dependencies so much that I don't understand why it's not the Gemfile default in official examples. You must ensure which version you're using and yes, I don't trust maintainers.

**Comment by enricostano** [comment]
> Are the three changes really related? Devise update, secret token configuration and session expiry? I don't see it.

Nope, I can separate this PR in 3 PRs if needed.

**Comment by enricostano** [comment]
> Final question: given that the session expiration is counted from the last request, isn't 1 hour a little too long? That is 1 hour of leaving the computer unattended. 

I put `1.hour` but we need to understand what's the right trade off between security and usability. Only time and people complains will tell the truth.

@sseerrggii any idea about this point?

File: Gemfile.lock (hunk 1 above)
Comment (rewritten) [inline Gemfile.lock]: Ok, let's go the hard version way, no problem. In any case `bundle outdated` would show us whether there is any update pending.

**Comment by rewritten** [comment]
so I think that two thirds of this can be merged right away. I start to see the point of very small PRs.

**Comment by enricostano** [comment]
> so I think that two thirds of this can be merged right away. I start to see the point of very small PRs.

Small PRs es bien :smile: 

Let's wait for some input on the expiration duration and then we'll merge this.

**Comment by sseerrggii** [comment]
I think 1 hour is a good solution :+1:

**Comment by rewritten** [comment]
Ok merging? or do we want to try and make tests for this? (My opinion is that testing this is part of the framework and we shouldn't test the framework)

**Comment by fcsonline** [comment]
:+1:

**Comment by enricostano** [comment]
OMG we have @fcsonline 's blessing! :scream:

**Comment by fcsonline** [comment]
hackers!!!

**Comment by enricostano** [comment]
BTW, we should change the `SECRET_TOKEN` in production to invalidate all the old cookies. Will do in provisioning repo.

**Pull request merged**